runtime/lua/vim/diagnostic.lua

```diff
@@ -447,7 +447,7 @@ local function set_list(loclist, opts)
     vim.fn.setqflist({}, ' ', { title = title, items = items })
   end
   if open then
-    vim.api.nvim_command(loclist and "lopen" or "copen")
+    vim.api.nvim_command(loclist and "lopen" or "botright copen")
```

Comment:
I think I'd rather add a `mods` field to the `opts` table that defaults to `botright` and use that.

```suggestion
    local mods = opts.mods or "botright"
    vim.api.nvim_command(mods .. " " .. (loclist and "lopen" or "copen"))
```

Reply:
Then the same value would apply to lopen and copen, the reason copen should have botright applied is because the quickfix menu is not window specific generally, so anchoring it's position to a given window leads to unintuitive behavior. I'm not sure we should make this configurable, otherwise for symmetry we would have to make the handlers for lsp configurable. I think having an opionated default here that aligns with the LSP handlers is fine.

Reply:
I agree an opinionated default is a good idea, I’m saying it should still be configurable for the cases when people don’t want to use the default.

Another option would be to fire a `QuickfixCmdPost` event that people can hook into to open the list however they want. I can do that in a separate PR though.

Reply:
If it is configurable, it should be per location list/copen, or perhaps open should itself be a boolean or string value, where string value is either lopen/copen/botrightcopen, or a function that can instead be overridden, not sure.

Reply:
Anyways, IMO we should merge this to bring all handlers/diagnostics in line and in a separate PR we can have a unified interface. There has been a lot of discussion/some PRs around minor additions to configurability of opening qfixlist/loclist (see https://github.com/neovim/neovim/pull/17339 for a recent one) so it would be good to figure out something uniform and update everyone.